### PR TITLE
Added documentation for VMware Photon Controller

### PR DESCRIPTION
**Pull request by AlainRoy**
This includes two things:
1. Documentation for kube-up integration with VMware's Photon Controller
2. Minor tweaks to vSphere documentation:
   - Changed maintainer
   - Added small note on vSphere parameters

### CODE REVIEW

**Comment by AlainRoy** [comment]
Please note that this documentation supports a change just made in Kubernetes:
https://github.com/kubernetes/kubernetes/pull/24124

**Comment by imkin** [comment]
This looks good to me.

**Comment by AlainRoy** [comment]
Is there anything I can do to aid the review process? I'm interested in getting this committed, but not sure what how to help it through.

**Comment by AlainRoy** [comment]
This pull request was made about six weeks ago but hasn't had any comments nor been merged. Is there something I can do to help? Did I make any mistakes in creating the pull request that make it unlikely to be merged?

**Comment by pwittrock** [comment]
@AlainRoy I am so sorry no one has reviewed this yet.  I just saw this pinged in the slack channel.

Can we file issues against you for this document if they are found?  Are the folks listed in the table willing to keep the guide up to date?

I ask because we are currently struggling with keeping the cluster guides fresh and well supported.

I will rely on your domain expertise for this guide and see if John can take a look asap.

**Comment by pwittrock** [comment]
@AlainRoy Would you be willing to push this to a github page so we can review the rendered docs.

http://kubernetes.io/editdocs/

Staging the site on GitHub Pages

**Comment by AlainRoy** [comment]
Yes you can file issues against me. Yes, I'll keep the guide up to date. Thanks for reviewing. I'll see if I can get it up on GitHub Pages soon.

**Comment by AlainRoy** [comment]
I was able to make alainroy.github.io, but it shows what's in my master branch, not what's in the branch I made. I normally keep the master branch pristine. What's the right thing to do? Just merge my changes into my repo's master branch? Something else?

**Comment by johndmulhausen** [comment]
@AlainRoy - Merge the changes into the master branch; you can also use that master branch as the base for a PR to here, where we also use the master branch. Sorry to shake up your conventions; the problem is GitHub Pages ONLY stages the master branch to username.github.io :/

Thanks a ton!

File: docs/getting-started-guides/photon-controller.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+Controller](http://vmware.github.io/photon-controller/)
+deployment. (Administrator access is only required for the initial
+setup: the actual creation of the cluster can be done by anyone.)
+
+2. The [Photon Controller CLI](https://github.com/vmware/photon-controller-cli)
+needs to be installed on the machine on which you'll be running kube-up. If you
+have go installed, this can be easily installed with:
+
+```
+    go get github.com/vmware/photon-controller-cli
```
Comment (johndmulhausen) [inline docs/getting-started-guides/photon-controller.md]: No leading whitespace

File: docs/getting-started-guides/photon-controller.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+1. You need administrator access to a [VMware Photon
+Controller](http://vmware.github.io/photon-controller/)
+deployment. (Administrator access is only required for the initial
+setup: the actual creation of the cluster can be done by anyone.)
+
+2. The [Photon Controller CLI](https://github.com/vmware/photon-controller-cli)
+needs to be installed on the machine on which you'll be running kube-up. If you
+have go installed, this can be easily installed with:
+
+```
```
Comment (johndmulhausen) [inline docs/getting-started-guides/photon-controller.md]: 22: ```shell

File: docs/getting-started-guides/photon-controller.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    go get github.com/vmware/photon-controller-cli
+```
+
+3. `mkisofs` needs to be installed. The installation process creates a
+CD-ROM ISO image to bootstrap the VMs with cloud-init. If you are on a
+Mac, you can install this with [brew](http://brew.sh/):
+
+```
+    brew install cdrtools
```
Comment (johndmulhausen) [inline docs/getting-started-guides/photon-controller.md]: No leading whitespace

File: docs/getting-started-guides/photon-controller.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+```
+    go get github.com/vmware/photon-controller-cli
+```
+
+3. `mkisofs` needs to be installed. The installation process creates a
+CD-ROM ISO image to bootstrap the VMs with cloud-init. If you are on a
+Mac, you can install this with [brew](http://brew.sh/):
+
+```
```
Comment (johndmulhausen) [inline docs/getting-started-guides/photon-controller.md]: 30: ```shell

File: docs/getting-started-guides/photon-controller.md [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+5. You should have an ssh public key installed. This will be used to
+give you access to the VM's user account, `kube`.
+
+6. Get or build a [binary release](/docs/getting-started-guides/binary_release)
+
+### Download VM Image
+
+Download a prebuilt Debian 8.2 VMDK that we'll use as a base image:
+
+```
```
Comment (johndmulhausen) [inline docs/getting-started-guides/photon-controller.md]: 45: ```shell

File: docs/getting-started-guides/photon-controller.md [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+When you do this, you'll need to configure the
+`cluster/photon-controller/config-common.sh` file with the names of
+the tenant, project, flavors, and image.
+
+If you prefer, you can  use the provided `cluster/photon-controller/setup-prereqs.sh`
+script to create these. Assuming the IP address of your Photon
+Controller is 192.0.2.2 (change as appropriate) and the downloaded image is in
+kube.vmdk, you can run:
+
+```
```
Comment (johndmulhausen) [inline docs/getting-started-guides/photon-controller.md]: 74: ```shell

File: docs/getting-started-guides/photon-controller.md [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+To create your Kubernetes cluster we will run the standard `kube-up`
+command. As described above, the parameters that control kube-up's
+interaction with Photon Controller are specified in files, not on the
+command-line.
+
+The time to deploy varies based on the number of nodes you create as
+well as the specifications of your Photon Controller hosts and
+network. Times vary from 10 - 30 minutes for a ten node cluster.
+
+```
```
Comment (johndmulhausen) [inline docs/getting-started-guides/photon-controller.md]: 112: ```shell

File: docs/getting-started-guides/photon-controller.md [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+Note that kube-up created a Kubernetes configuration file for you in
+`~/.kube/config`. This file will allow you to use the `kubectl`
+command. It contains the IP address of the Kubernetes master as well
+as the password for the `admin` user. If you wish to use the
+Kubernetes web-based user interface you will need this password. In
+the config file you'll see a section that look like the following: you
+use hte password there. (Note that the output has been trimmed: the
+certificate data is much lengthier)
+
+```
```
Comment (johndmulhausen) [inline docs/getting-started-guides/photon-controller.md]: 128: ```yaml

File: docs/getting-started-guides/photon-controller.md [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+    client-key-data: LS0tL...
+    password: PASSWORD-HERE
+    username: admin
+```
+
+### Removing your Kubernetes cluster
+The recommended way to remove your Kubernetes cluster is with the
+`kube-down` command:
+
+```
```
Comment (johndmulhausen) [inline docs/getting-started-guides/photon-controller.md]: 141: ```shell

File: docs/getting-started-guides/photon-controller.md [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+One option is to use the NodePort option with a manually deployed
+balancer. Specifically:
+
+1. Configure your service with the NodePort option. For example, this
+service uses the NodePort option. All Kubernetes nodes will listen on
+a port and forward network traffic to any pods in the service. In this
+case, Kubernets will choose a random port, but it will be the same
+port on all nodes.
+
+```
```
Comment (johndmulhausen) [inline docs/getting-started-guides/photon-controller.md]: 164: ```yaml

File: docs/getting-started-guides/photon-controller.md [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+
+2. Create a new standalone VM (or VMs, for high availability) to act
+as a load balancer. For example, if you use haproxy, you could make a
+configuration similar to the one below. Note that this example assumes there
+are three Kubernetes nodes: you would adjust the configuration to reflect the
+actual nodes you have. Also note that port 30144 should be replaced
+with whatever NodePort was assigned by Kubernetes.
+
+```
```
Comment (johndmulhausen) [inline docs/getting-started-guides/photon-controller.md]: 188: ```shell

File: docs/getting-started-guides/photon-controller.md [hunk 12]
```diff
@@ -0,0 +1,227 @@
...
+* An image: we'll use the one above
+
+When you do this, you'll need to configure the
+`cluster/photon-controller/config-common.sh` file with the names of
+the tenant, project, flavors, and image.
+
+If you prefer, you can  use the provided `cluster/photon-controller/setup-prereqs.sh`
+script to create these. Assuming the IP address of your Photon
+Controller is 192.0.2.2 (change as appropriate) and the downloaded image is in
+kube.vmdk, you can run:
```
Comment (johndmulhausen) [inline docs/getting-started-guides/photon-controller.md]: `kube.vmdk`

**Comment by AlainRoy** [comment]
Thanks. I switched to my master branch, so I made a new pull request: 
https://github.com/kubernetes/kubernetes.github.io/pull/743

I'll close this one.

**Pull request closed without merging**